src/main/java/com/serhiiklymchuk/gmailpet/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.serhiiklymchuk.gmailpet.service.impl;
+
+import com.serhiiklymchuk.gmailpet.domain.User;
+import com.serhiiklymchuk.gmailpet.domain.UserRole;
+import com.serhiiklymchuk.gmailpet.dto.UserDto;
+import com.serhiiklymchuk.gmailpet.repository.UserRepository;
+import com.serhiiklymchuk.gmailpet.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService, UserDetailsService {
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository
+                .findByUsername(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User was not found!!"));
+    }
+
+    @Override
+    public void createUser(UserDto userDto) {
+        String encodedPassword = new BCryptPasswordEncoder().encode(userDto.getPassword());
+
+        User user = User.builder()
+                .firstName(userDto.getFirstName())
+                .lastName(userDto.getLastName())
+                .username(userDto.getUsername())
+                .password(encodedPassword)
+                .role(UserRole.ROLE_USER)
+                .active(true)
+                .build();
+
+        userRepository.save(user);
+    }
+
+    @Override
+    public User findByUsername(String username) {
```

Comment:
I see the same logic twice in this class `loadUserByUsername` and `findByUsername`. Remove this one